consul/autopilot.go

```diff
@@ -113,13 +82,18 @@ func (s *Server) pruneDeadServers() error {
 		}
 	}
 
+	// Nothing to remove, return early
+	if len(failed) == 0 {
+		return nil
+	}
+
 	peers, err := s.numPeers()
 	if err != nil {
 		return err
 	}
 
 	// Only do removals if a minority of servers will be affected
-	if len(failed) <= peers/2 {
+	if len(failed) < peers/2 || (len(failed) == 1 && peers >= 3) {
```

Comment:
I might not be thinking about the case you were trying to hit though, so definitely interested in your rationale.